packages/components/upload/src/upload-list.vue

```diff
@@ -107,67 +106,32 @@ import {
 import { useLocale, useNamespace } from '@element-plus/hooks'
 import ElProgress from '@element-plus/components/progress'
 
-import type { PropType } from 'vue'
-import type { UploadFile } from './upload.type'
+import { uploadListEmits, uploadListProps } from './upload-list'
+import type { UploadFile } from './upload'
 
-export default defineComponent({
+defineOptions({
   name: 'ElUploadList',
-  components: {
-    ElProgress,
-    ElIcon,
-    Document,
-    Delete,
-    Close,
-    ZoomIn,
-    Check,
-    CircleCheck,
-  },
-  props: {
-    files: {
-      type: Array as PropType<UploadFile[]>,
-      default: () => [] as File[],
-    },
-    disabled: {
-      type: Boolean,
-      default: false,
-    },
-    handlePreview: {
-      type: Function as PropType<(file: UploadFile) => void>,
-      default: () => NOOP,
-    },
-    listType: {
-      type: String as PropType<'picture' | 'picture-card' | 'text'>,
-      default: 'text',
-    },
-  },
-  emits: ['remove'],
-  setup(props, { emit }) {
-    const { t } = useLocale()
-    const nsUpload = useNamespace('upload')
-    const nsIcon = useNamespace('icon')
-    const nsList = useNamespace('list')
+})
 
-    const handleClick = (file: UploadFile) => {
-      props.handlePreview(file)
-    }
+const props = defineProps(uploadListProps)
+const emit = defineEmits(uploadListEmits)
 
-    const onFileClicked = (e: Event) => {
-      ;(e.target as HTMLElement).focus()
-    }
+const { t } = useLocale()
+const nsUpload = useNamespace('upload')
+const nsIcon = useNamespace('icon')
+const nsList = useNamespace('list')
 
-    const handleRemove = (file: UploadFile) => {
-      emit('remove', file)
-    }
-    return {
-      focusing: ref(false),
-      handleClick,
-      handleRemove,
-      onFileClicked,
-      t,
-      nsUpload,
-      nsIcon,
-      nsList,
-    }
-  },
-})
+const focusing = ref(false)
+
+const handleClick = (file: UploadFile) => {
```

Comment:
**Question:** can we just use `props.handlePreview`?

Reply:
I don't recommend this, external APIs should not be passed directly to internal.